### PR TITLE
1. Add Op class support for multi input and multi output function. 2.…

### PR DESCRIPTION
**Pull request by gongshaotian**
… Add DRR duplicate TransposeOp merge testing code

### PR types

Othres

### PR changes

Othres

### Description

1. Add Op class support for multi input and multi output function. 
2. Add DRR duplicate TransposeOp merge testing code

### CODE REVIEW

File: paddle/ir/pattern_rewrite/drr/api/drr_pattern_context.h [hunk 1]
```diff
@@ -187,18 +188,21 @@ class OpCall {
...
-  const std::string& name() const { return op_->name(); }
+  const std::string& name() const { return op_type_name_; }
 
   const std::vector<const Tensor*>& inputs() const { return inputs_; }
 
   const std::vector<const Tensor*>& outputs() const { return outputs_; }
 
  private:
-  const Op* op_;
+  // const Op* op_;
```
Comment (zyfncg) [inline paddle/ir/pattern_rewrite/drr/api/drr_pattern_context.h]: 注释可以删掉了

File: paddle/ir/pattern_rewrite/drr/api/drr_pattern_context.h [hunk 2]
```diff
@@ -187,18 +188,21 @@ class OpCall {
...
+  std::string op_type_name_;
```
Comment (zyfncg) [inline paddle/ir/pattern_rewrite/drr/api/drr_pattern_context.h]: op_type_name_ -> op_name_

File: paddle/ir/pattern_rewrite/drr/api/drr_pattern_context.cc [hunk 3]
```diff
@@ -83,6 +83,12 @@ void Op::operator()(const Tensor& arg, const Tensor* out) const {
   pattern_graph_->AddOpCall(std::make_shared<OpCall>(this, inputs, outputs));
 }
 
+void Op::operator()(const std::vector<const Tensor*>& args, const std::vector<const Tensor*>& outputs) const{
```
Comment (zyfncg) [inline paddle/ir/pattern_rewrite/drr/api/drr_pattern_context.cc:86]: args 可以改成 const std::vector<const Tensor&>& 类型吗

File: paddle/ir/pattern_rewrite/drr/drr_rewrite_pattern.h [hunk 4]
```diff
@@ -54,13 +54,16 @@ class DrrRewritePattern : public ir::OpRewritePattern<SourceOp> {
...
-    auto* anchor = source_pattern_graph_->AnchorNode();
+    const auto* anchor = source_pattern_graph_->AnchorNode();
     IR_ENFORCE(anchor);
     std::unordered_set<const OpCall*> drr_visited;
     std::unordered_set<const Operation*> ir_visited;
     std::queue<const OpCall*> drr_q;
     std::queue<const ir::Operation*> ir_q;
     drr_q.push(anchor);
+
+    VLOG(0) << "anchor->name(): " << anchor->name();
```
Comment (zyfncg) [inline paddle/ir/pattern_rewrite/drr/drr_rewrite_pattern.h]: 调试日志都删一下

File: paddle/ir/pattern_rewrite/drr/api/drr_pattern_context.h (hunk 2 above)
Comment (gongshaotian) [reply paddle/ir/pattern_rewrite/drr/api/drr_pattern_context.h]: ok

File: paddle/ir/pattern_rewrite/drr/api/drr_pattern_context.h (hunk 1 above)
Comment (gongshaotian) [reply paddle/ir/pattern_rewrite/drr/api/drr_pattern_context.h]: 已删除

File: paddle/ir/pattern_rewrite/drr/api/drr_pattern_context.cc (hunk 3 above)
Comment (gongshaotian) [reply paddle/ir/pattern_rewrite/drr/api/drr_pattern_context.cc:86]: 可以改，但开销比较大

**Comment by gongshaotian** [comment]
1. Add transposeOp in createOption func
2. Add output tensor judgment for graphs in the match method
3. Delete VELOG in the match method

**Pull request closed without merging**